src/Main.hs

```diff
@@ -71,10 +71,9 @@ trace :: Int -> Int -> Color
 trace x y =
     let
       ray =  generateRay camera sceneWidth sceneHeight x y
-      intersections = closestIntersection ray objects
+      intersection = closestIntersection ray objects
     in
-      case intersections of Nothing -> Color.white
-                            Just distObj -> getColorFromIntersection distObj
+      maybe Color.white getColorFromIntersection intersection
```

Comment:
this is essentially the same as mapping a function over a maybe, then doing fromMaybe with a default. just does it in one function.